aws_jupyter_proxy/awsproxy.py

```diff
@@ -224,6 +225,18 @@ def _compute_downstream_headers(self, downstream_request_path) -> HTTPHeaders:
         base_service_url = urlparse(self.service_info.endpoint_url)
         downstream_request_headers["Host"] = base_service_url.netloc
 
+        user_agent_product = "aws-jupyter-proxy"
+        user_agent_version = pkg_resources.require("aws_jupyter_proxy")[0].version
+        user_agent = user_agent_product + "/" + user_agent_version
+
```

Comment:
How expensive is computing this on every request? This isn't something that should change often so would a simple `@lru_cache` be beneficial here?

Reply:
I decided to declare a global variable on the module level to store the version number since the value remains the same on each server startup and would not need a cache replacement strategy.